### DISCUSSION TITLE
Singular shape tensor in Peridynamics

### DISCUSSION THREAD

**Discussion by Yunlong010413** [idaholab/moose#24004] [Q&A Modules: General]
I'm a newbie, but I'm currently experiencing the following problems.What can I do if the following problems occur? Thank you.
```
*** ERROR ***
The following error occurred in the object "strain", of type "ComputePlaneSmallStrainNOSPD".
Singular shape tensor is detected in ComputeStrainBaseNOSPD! Use SingularShapeTensorEliminatorUserObjectPD to avoid singular shape tensor!
```
I've tried everything I can, but to no avail, it seems that I can only modify the number of grids to work properly. This is an example from the most common tests, just increasing the number of meshes, but this problem occurs. Is there any other solution?

**Comment by lindsayad** [comment]
@hchen139 do you have any suggestions for this user?

**Reply by GiudGiud** [reply]
Is this an issue with uniform refinement?

**Reply by hchen139** [reply]
I don't think PD uses any refinement. Proper use of the 'SingularShapeTensorEliminatorUserObjectPD' should be able to avoid the singularity issue.

**Comment by andrisfreimanis** [comment]
I had a similar issue a while ago. The problem was that the SingularShapeTensorEliminatorUserObjectPD was not the first user object executed. Maybe this helps https://github.com/idaholab/moose/issues/24239
